Code/Tools/FBuild/FBuildCore/Graph/VCXProjectNode.cpp

```diff
@@ -63,6 +63,12 @@ VCXProjectNode::VCXProjectNode( const AString & projectOutput,
 
     // depend on the input nodes
     m_StaticDependencies.Append( paths );
+
+    // generate GUID if not specified
+    if ( m_ProjectGuid.IsEmpty() )
+    {
+        VSProjectGenerator::FormatDeterministicProjectGUID( m_ProjectGuid, m_Name );
```

Comment:
I think we need to truncate the path to be relative to the working directory so that the same GUID is generated for two different users who might have different paths.